sympy/core/numbers.py

```diff
@@ -222,6 +223,66 @@ def _literal_float(f):
     """Return True if n starts like a floating point number."""
     return bool(_floatpat.match(f))
 
+
+def _most_sig(x, prec=None):
+    # return 0 if mpf precision < 1
+    if x[3] < 1:
+        return 0
+
+    x = (0,) + x[1:]
+
+    if not prec:
+        prec = x[3]
+    prec += 20
+
+    decimal_digits = mlib.to_int(mlib.mpf_floor(mlib.mpf_div(
+        mlib.mpf_log(x, prec), mlib.mpf_ln10(prec), prec
+    )))
+
+    binary_prec = math.copysign(abs(dps_to_prec(decimal_digits + 1)), decimal_digits)
+
+    return int(binary_prec - 4)
```

Comment:
This whole block is confusing. It seems to:

1. add 20 to prec
2. convert magnitude of x relative to prec in decimal digits using floating point
3. add 1 to decimal prec
4. convert prec to binary
5. subtract 4 from prec 

If this is being called as part of e.g. `__add__` then I don't think that there should be any conversion to decimal here.

What exactly is this doing that can't be done using integer calculations with the binary precision and the components of the `_mpf_`?